client/src/main/java/io/lionweb/client/inmemory/InMemoryServer.java

```diff
@@ -182,6 +229,53 @@ public Map<String, ClassifierResult> nodesByLanguage(
     return result;
   }
 
+  //
+  // Delta methods
+  //
+
+  public void monitorDeltaChannel(String repositoryName, @NotNull DeltaChannel channel) {
+    Objects.requireNonNull(channel, "Channel should not be null");
+    channel.registerCommandReceiver(new DeltaCommandReceiverImpl(repositoryName, channel));
+  }
+
+  private class DeltaCommandReceiverImpl implements DeltaCommandReceiver {
+    private String repositoryName;
+    private DeltaChannel channel;
+
+    private DeltaCommandReceiverImpl(String repositoryName, DeltaChannel channel) {
+      this.repositoryName = repositoryName;
+      this.channel = channel;
+    }
+
+    @Override
+    public DeltaCommandResponse receiveCommand(DeltaCommand command) {
+      if (command instanceof ChangeProperty) {
+        ChangeProperty changeProperty = (ChangeProperty) command;
+        RepositoryData repositoryData = getRepository(repositoryName);
+        List<SerializedClassifierInstance> retrieved = new ArrayList<>();
+        try {
+          repositoryData.retrieve(changeProperty.node, 0, retrieved);
+        } catch (IllegalArgumentException e) {
+          return DeltaCommandResponse.rejected(
+              "Node with id " + changeProperty.node + " not found");
+        }
+        SerializedClassifierInstance node = retrieved.get(0);
+        String oldValue = node.getPropertyValue(((ChangeProperty) command).property);
```

Comment:
I tried to separate the protocol handling from _applying_ the deltas to a local model, and this turned out to be helpful: In our project, we're actually not using the repository, but recording all deltas on one side, sending them to the other side, and apply them there. We can reuse all the logic written for repository.